PR: https://github.com/odoo/odoo/pull/74967

From: dbe22a3c06a5e25f28fd33ef5ca24e4f86ebb327
From: Nidhi Patel
Date: 2021-09-03 15:45:14

Structural Changes: 6
Total Changes: 6

[IMP] {hr,sale}_timesheet: improve setting in the timesheet module

The purpose of this task is to improve the setting of the timesheet module.

In this commit, we improve the following:
 - Rename the label, description and align the field with the name and
   description of the feature
 - Add an 'OK' button in the 'download our app' modal
 - Render the frequency field properly in the employee/manager reminders
   setting on mobile view
 - In configure your services link: change the name of the action to "Services"
 - Display the following options in the community, with the usual 'Enterprise'
   tag:  Round Timesheet, Employee/Manager reminder, Invoicing Policy

closes odoo/odoo#74967

Task-id: 2531338
Related: odoo/enterprise#20182
Related: odoo/upgrade#2771
Signed-off-by: LTU-Odoo <IT-Ideas@users.noreply.github.com>

================================= pseudo patch: =================================

--- a/addons/hr_timesheet/models/res_config_settings.py
+++ b/addons/hr_timesheet/models/res_config_settings.py
@@ -9,8 +9,10 @@ class ResConfigSettings(models.TransientModel):
 
     module_project_timesheet_synchro = fields.Boolean("Awesome Timesheet",
         compute="_compute_timesheet_modules", store=True, readonly=False)
-    module_project_timesheet_holidays = fields.Boolean("Record Time Off",
+    module_project_timesheet_holidays = fields.Boolean("Time Off",
         compute="_compute_timesheet_modules", store=True, readonly=False)
+    reminder_user_allow = fields.Boolean(string="Employee Reminder", help="If checked, send an email to all users who have not recorded their timesheet")
+    reminder_manager_allow = fields.Boolean(string="Manager Reminder", help="If checked, send an email to all manager")
     project_time_mode_id = fields.Many2one(
         'uom.uom', related='company_id.project_time_mode_id', string='Project Time Unit', readonly=False,
         help="This will set the unit of measure used in projects and tasks.\n"
@@ -20,8 +22,6 @@ class ResConfigSettings(models.TransientModel):
         related='company_id.timesheet_encode_uom_id', readonly=False,
         help="""This will set the unit of measure used to encode timesheet. This will simply provide tools
         and widgets to help the encoding. All reporting will still be expressed in hours (default value).""")
-    timesheet_min_duration = fields.Integer('Minimal duration', default=15, config_parameter='hr_timesheet.timesheet_min_duration')
-    timesheet_rounding = fields.Integer('Rounding up', default=15, config_parameter='hr_timesheet.timesheet_rounding')
     is_encode_uom_days = fields.Boolean(compute='_compute_is_encode_uom_days')
 
     @api.depends('timesheet_encode_uom_id')
